proposals/partial-type-inference.md

```diff
@@ -0,0 +1,738 @@
+  -   | 'var'    | 'when'       | 'where'     | 'yield'
+    ;
+  ```
+
+### Type arguments
+
+> Specification: Original section changed in the following way
+
+* We change the meaning of the content of *type_argument_list* in two contexts.
+  * [Constructed types](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/types.md#84-constructed-types) occuring in [*object_creation_expression*](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/expressions.md#128162-object-creation-expressions)
+  * Constructed types and type arguments occuring in method [invocation](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/expressions.md#12892-method-invocations)
+
+* ***inferred_type_argument*** represents an unknown type, which will be resolved during type inference. 
+
+* `_` identifier is considered to represent *inferred_type_argument* when:
+  * It occurs in *type_argument_list* of a method group during method invocation.
+  * It occurs in *type_argument_list* of a type in *object_creation_expression*.
+  * It occurs as an arbitrary nested identifier in the expressions mentioned above.
+  > Example
+  >
+  > ```csharp
+  > F<_, int>( ... ); // _ represents an inferred type argument.
+  > new C<_, int>( ... ); // _ represents an inferred type argument.
+  > F<C<_>, int>( ... ); // _ represents an inferred type argument.
+  > new C<C<_>, int>( ... ); // _ represents an inferred type argument.
+  > C<_> temp = ...; // _ doesn't represent an inferred type argument.
+  > new _( ... ) // _ doesn't represent an inferred type argument.
+  > Container<_>.Method<_>(arg); // _ of Container<_> doesn't represent an inferred type argument. (Containing type's type argument won't be inferred)
+  > ```   
+
+* A method group and type are said to be *partial_inferred* if it contains at least one *inferred_type_argument*. 
+
+* A type is said to be *generic_inferred* when all the following hold:
+  * It has an empty *type_argument_list*.
+  * It occurs as a *type* of *object_creation_expression*.
+  > Example
+  >
+  > ```csharp
+  > new C<>(...) // Valid code, C is generic_inferred.
+  > new C<G<>>(...) // Invalid code, C nor G are generic_inferred.
+  > F<>(...) // Invalid code, F isn't generic_inferred.
+  > ```
+
+### Namespace and type names
+
+> Specification: Original section changed in the following way
+
+Determining the [meaning](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/basic-concepts.md#781-general) of a *namespace_or_type_name* is changed as follow.
+
+* If a type is a *generic_inferred*, then we resolve the identifier in the same manner except ignoring the arity of type parameters (Types of arity 0 is ignored). 
+If there is an ambiguity in the current scope, a compilation-time error occurs.
+  > Example
+  >
+  > ```csharp
+  > class P1
+  > {
+  >     void M() 
+  >     {
+  >         new C1<>( ... ); // Refers generic_inferred type C1<T>
```

Comment:
`T`, `T1` and `T2` don't exist here.
Reading ahead to the "Object creation expressions" section, I believe the comments in this snippet are not what you intended and some actual types should be inferred from inspecting candidate constructors.

Reply:
Should be fixed by commit [b6081e3](https://github.com/dotnet/csharplang/pull/7582/commits/b6081e37e9a7e3b6892d3b0164e5d91b3a61cfc3)